### PR TITLE
Implement filtering

### PR DESCRIPTION
**Pull request by RobStafford**
This was a fairly major refactor because I had to push state back up to Ratings (I had moved much of it to Ratings List). This required changing two different components to stateful, and passing an immense amount of data & functions down, but everything should be working.

FYI: this PR should pass tests, but my last serious piece of technical debt (which will be the next PR) is the new review screen, where I will be turning on picture upload, which will break the tests.

Cheers!

### CODE REVIEW

**Comment by RobStafford** [comment]
OK, this WOULD have passed all checks if it wasn't for QA (& maybe Overview) :-p. I promise nothing is broken.

File: client/src/components/Ratings/Ratings.jsx [hunk 1]
```diff
@@ -115,14 +168,76 @@ class Ratings extends React.Component {
...
 
     return (
       <div data-testid="ratings" className="ratings-widget">
         <a id="reviews-link"></a>
         <h3>RATINGS & REVIEWS</h3>
-        <br></br><RatingsMeta className="ratings_meta" ratings_meta={this.state.ratings_meta}></RatingsMeta><RatingsList className="ratings_list" ratings_meta={this.state.ratings_meta} 
-        count={this.state.count} item_name={this.state.item_name} chooseHelpful={this.chooseHelpful} putMarkHelpful={this.putMarkHelpful} getAllReviews={this.getAllReviews.bind(this)}></RatingsList>
+        <br></br><RatingsMeta item_id={this.state.item_id} sort={this.state.sort} filters={this.state.filters} className="ratings_meta" ratings_meta={this.state.ratings_meta} getAllReviews={this.getAllReviews.bind(this)} clearFilters={this.clearFilters} count={this.state.count} flip_filters={this.flipFilters} filter_data={this.filterData} any_filters={this.state.any_filters} ></RatingsMeta><RatingsList className="ratings_list" updateDisplayedItems={this.updateDisplayedItems} moreReviews={this.moreReviews} handleSort={this.handleSort} ratings_to_display={this.state.ratings_to_display} ratings_meta={this.state.ratings_meta} filters={this.state.filters}
```
Comment (DominicMonares) [inline client/src/components/Ratings/Ratings.jsx]: line should be split

File: client/src/components/Ratings/Ratings.jsx [hunk 2]
```diff
@@ -115,14 +168,76 @@ class Ratings extends React.Component {
...
+        count={this.state.count} count_to_display={this.state.count_to_display} item_name={this.state.item_name} chooseHelpful={this.chooseHelpful} putMarkHelpful={this.putMarkHelpful} getAllReviews={this.getAllReviews.bind(this)} all_reviews={this.state.ratings}></RatingsList>
```
Comment (DominicMonares) [inline client/src/components/Ratings/Ratings.jsx]: line should be split

File: client/src/components/Ratings/components/RatingsList.jsx [hunk 3]
```diff
@@ -77,7 +76,7 @@ class RatingsList extends React.Component {
       }
     }
 
-    let ratings = this.state.ratings_to_display.map((rating) => {
+    let ratings = this.props.ratings_to_display.map((rating) => {
       return (
           <IndividualReview chooseHelpful={chooseHelpful} putMarkHelpful={putMarkHelpful} key={rating.review_id} review_id={rating.review_id} star_rating={rating.rating} summary={rating.summary} date={rating.date} body={rating.body} recommend={rating.recommend} reviewer_name={rating.reviewer_name} response={rating.response} helpfulness={rating.helpfulness} photos={rating.photos}></IndividualReview>
```
Comment (DominicMonares) [inline client/src/components/Ratings/components/RatingsList.jsx]: line should be split

**Review by DominicMonares — CHANGES_REQUESTED** [review]
Looks great overall! Just need a few line splits and it's an approval from me.

**Comment by RobStafford** [comment]
I have split up the lines. :-)

**Pull request merged**